### PR TITLE
Improve checkpointing docs

### PR DESCRIPTION
**Pull request by szkarpinski**
## Category:
**Other** - Documentation

## Description:

- Added a small example of checkpointing usage to pipeline docs
- Fixed typos in checkpointing docs and removed out of date statements about checkpointning only at the beginning of an epoch
- Fixed a mistake in `enable_memory_stats` docs

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by mzient — APPROVED** [review]

File: dali/python/nvidia/dali/pipeline.py [hunk 1]
```diff
@@ -108,14 +108,28 @@ class Pipeline(object):
...
-    from the saved iteration.
+    from the saved iteration::
+
+        @pipeline_def(..., enable_checkpointing=True)
+        def pipeline():
+            ...
+        
+        p = pipeline()
+        p.build()
+        # Run the pipeline...
```
Comment (stiepan) [inline dali/python/nvidia/dali/pipeline.py:126]: nitpick: At first I thought the `Run the pipeline` refers to the line below, where we save the checkpoint. Maybe adding p.run() or even some for loop to make the example more explicit would help disambiguate it?

Or mabe it's just me.

Comment (szkarpinski) [reply dali/python/nvidia/dali/pipeline.py]: Good point, I changed it to explicitly running p.run()

**Review by awolant — APPROVED** [review]

**Comment by szkarpinski** [comment]
!build

**Comment by szkarpinski** [comment]
!build

**Comment by szkarpinski** [comment]
!build

**Pull request merged**